### PR TITLE
fix: use correct type for '*?enhanced' imports

### PR DESCRIPTION
**Pull request by ranjan-purbey**
---

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [ ] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### Edits

- [x] Please ensure that 'Allow edits from maintainers' is checked. PRs without this option may be closed.

Currently the type of an enhanced import is `string`, which is wrong. You can easily verify by logging the imported value. The correct type is `Picture`.

### CODE REVIEW

File: .changeset/cool-grapes-remain.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@sveltejs/enhanced-img": patch
+---
+
+Use correct type for '\*?enhanced' imports
```
Comment (benmccann) [inline .changeset/cool-grapes-remain.md:5]: fix: use correct type for `*?enhanced` imports
```

File: packages/enhanced-img/types/ambient.d.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
+import type { Picture } from 'vite-imagetools';
```
Comment (benmccann) [inline packages/enhanced-img/types/ambient.d.ts:1]:
```suggestion
import type { Picture } from 'vite-imagetools';

```

File: packages/enhanced-img/types/ambient.d.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
...
 declare module '*?enhanced' {
-	const value: string;
+	const value: Picture;
```
Comment (benmccann) [inline packages/enhanced-img/types/ambient.d.ts:3]:
```suggestion
	const value: string | Picture;
```

File: .changeset/cool-grapes-remain.md (hunk 1 above)
Comment (benmccann) [inline .changeset/cool-grapes-remain.md:5]:
```suggestion
fix: use correct type for `*?enhanced` imports
```

File: packages/enhanced-img/types/index.d.ts [hunk 4]
```diff
@@ -1,11 +1,14 @@
+import 'svelte/elements';
```
Comment (benmccann) [inline packages/enhanced-img/types/index.d.ts:1]: I'm not much of a typescript expert. why was this changed from `import { HTMLImgAttributes } from 'svelte/elements';`?

Comment (ranjan-purbey) [reply packages/enhanced-img/types/index.d.ts:1]: Since `HTMLImgAttributes` is not used anywhere else outside the declare block, typescript compains that it is an unused import. (similar to https://github.com/microsoft/TypeScript/issues/45061)

So one way to solve this is to import the module itself. Other option is to declare a type alias outside the `declare` block. Third option is to use the dynamic type import syntax.

**Review by benmccann — APPROVED** [review]

**Comment by itsmikesharescode** [comment]
this looks broken e-e

**Comment by benmccann** [comment]
How do?

**Comment by BastiDood** [comment]
> this looks broken e-e

This should be fixed by #12363. Cc: @benmccann
